mpas_analysis/ocean/conservation.py

```diff
@@ -0,0 +1,633 @@
+            Contains configuration options
+        """
+        # Authors
+        # -------
+        # Carolyn Begeman
+
+        super(ConservationTask, self).__init__(
+            config=config,
+            taskName='oceanConservation',
+            componentName='ocean',
+            tags=['timeSeries', 'conservation'])
+
+    def setup_and_check(self):
+        """
+        Perform steps to set up the analysis and check for errors in the setup.
+        Also, adds attributes to the task that will be needed later.
+
+        Raises
+        ------
+        ValueError: if plot type is not supported, conservation analysis member
+            is inactive or input files are missing
+        """
+        # Authors
+        # -------
+        # Carolyn Begeman
+
+        super(ConservationTask, self).setup_and_check()
+
+        # Check that the conservation analysis member is active
+        self.check_analysis_enabled(
+            analysisOptionName='config_am_conservationcheck_enable',
+            raiseException=True)
+
+        # Specify where to put analysis task output
+        config = self.config
+        baseDirectory = build_config_full_path(
+            config, 'output', 'conservationSubdirectory')
+        make_directories(baseDirectory)
+        self.outputFile = f'{baseDirectory}/{self.fullTaskName}.nc'
+
+        # get a list of conservationCheck output files from the streams file,
+        # reading only those that are between the start and end dates
+
+        # the run directory contains the restart files
+        self.runDirectory = build_config_full_path(self.config, 'input',
+                                                   'runSubdirectory')
+        # if the history directory exists, use it; if not, fall back on
+        # runDirectory
+        self.historyDirectory = build_config_full_path(
+            self.config, 'input',
+            f'{self.componentName}HistorySubdirectory',
+            defaultPath=self.runDirectory)
+
+        self.startYear = self.config.getint('conservation', 'startYear')
+        self.endYear = self.config.getint('conservation', 'endYear')
+        self.inputFiles = sorted(self.historyStreams.readpath(
+            'conservationCheckOutput',
+            startDate=f'{self.startYear:04d}-01-01_00:00:00',
+            endDate=f'{self.endYear:04d}-01-01_00:00:00',
+            calendar=self.calendar))
+
+        if len(self.inputFiles) == 0:
+            raise IOError(f'No files were found matching {self.inputFiles}')
+
+        with xr.open_dataset(self.inputFiles[0]) as ds:
+            self.allVariables = list(ds.data_vars.keys())
+
+        self.mainRunName = self.config.get('runs', 'mainRunName')
+        self.referenceRunName = \
+            config.get('runs', 'preprocessedReferenceRunName')
+        self.referenceInputDirectory = config.get('oceanPreprocessedReference',
+                                                  'baseDirectory')
```

Comment:
These config options related to "preprocessed reference runs" are for a different purpose.  We no longer use that feature but it was needed in the early days of MPAS-Analysis to compare E3SM v0 simulations against CESM data that had been processed to "look" similar.

The way that this should be handled instead is by passing a `controlConfig` to the constructor of the class and then, if `self.controlConfig is not None`, do the steps needed to load the corresponding data sets from the control analysis run.

In your main config file, all that should be needed is to point to the control config file.  No other config options should need to be supplied.

Reply:
Got it. Thanks!